base/subarray.jl

```diff
@@ -123,8 +123,9 @@ _maybe_reshape_parent(A::AbstractArray, ::NTuple{N, Bool}) where {N} = reshape(A
     view(A, inds...)
 
 Like [`getindex`](@ref), but returns a view into the parent array `A` with the
-given indices instead of making a copy.  Calling [`getindex`](@ref) or
-[`setindex!`](@ref) on the returned `SubArray` computes the
+given indices instead of making a copy, unless `A` is immutable.  
```

Comment:
Kinda nitpick, but a vector can be (im)mutable in at least two senses: `setindex!` works or `push!` works. I think we'd only want to guarantee that `setindex!(A, ...)`  works => `setindex!(view(A, ...), ...)`  works.

```suggestion
given indices instead of making a copy if (but not only if) `A` supports `setindex!`.
Otherwise, it may create a new object if that is a cheap operation (e.g., `UnitRange`).
```